lib/tools/system-calls.js

```diff
@@ -84,11 +83,17 @@ systemCallMethods.getBinaryFromSdkRoot = _.memoize(async function (binaryName) {
                     path.resolve(this.sdkRoot, "tools", binaryName),
                     path.resolve(this.sdkRoot, "tools", "bin", binaryName)];
   // get subpaths for currently installed build tool directories
-  let buildToolDirs = await getDirectories(path.resolve(this.sdkRoot, "build-tools"));
-  // the newest version goes first
-  buildToolDirs.sort(semver.rcompare);
-  for (let versionDir of buildToolDirs) {
-    binaryLocs.push(path.resolve(this.sdkRoot, "build-tools", versionDir, binaryName));
+
+  let buildToolDirs = await fs.glob(path.resolve(this.sdkRoot, 'build-tools', '*') + path.sep, {absolute: true});
+  if (buildToolDirs.length) {
+    const pairs = await B.map(buildToolDirs, async (dir) => [(await fs.stat(dir)).mtimeMs, dir]);
+    buildToolDirs = pairs
+      .sort((a, b) => a[0] > b[0])
+      .map((pair) => pair[1]);
+    log.info(`Found the following 'build-tools' folders (the newest comes first): ${buildToolDirs}`);
+    buildToolDirs.map(x => binaryLocs.push(path.resolve(x, binaryName)));
```

Comment:
I don't think `map` is the right function here, since you are just using it to iterate over the array. `_.forEach` would make more sense.

Reply:
Also, given that the `binaryLocs` is being updated, the error message when nothing is found should be changed.

Reply:
which error message you would like to see?

Reply:
Well, the error message at https://github.com/mykola-mokhnach/appium-adb/blob/df7d0ba95b8f00fb3a5f722f78ec26094958123b/lib/tools/system-calls.js#L105-L107 will say `Could not find ${binaryName} in ${binaryLocs}, or supported build-tools under ${this.sdkRoot} do you have the Android SDK installed at this location?`, but `binaryLocs` will include all the build-tools directories, so it is redundant.

Reply:
it might be still useful to see which binary version was used later

Reply:
I agree. This is just about the error, which is just redundant now. I like the inclusion of the information about the one used in the success case.